yarn_api_client/base.py

```diff
@@ -12,37 +12,57 @@ def __init__(self, response):
         self.data = response.json()
 
 
+class Uri(object):
```

Comment:
Please use the existing Uri functionality `from .uri import Uri` which will allow you to use the code snip described in #33

Reply:
The problem with it is that user will be more limited with types of format for URL. I tried multiple ways, but method `.getUrl()` doesn't prepend defaunt protocol and port

Reply:
I agree that we should not perform our own parsing of `service_endpoint`.  (Btw, @lresende's code snippet for his `Uri` class is just a wrapper of `urlparse` that "converts" `path` to `api_endpoint`.)

```python
service_uri = Uri(service_endpoint)
self.scheme = service_uri.scheme or 'https'
self.hostname = service_uri.hostname
self.port = service_uri.port or '8088'
self.api_endpoint = service_uri.path or '/ws/v1/cluster'
```
@dimon222 - how does this not address defaulting of the protocol (scheme) or port?  Can you please clarify?

Btw, I think we'd want to the default scheme to be `http` since the port is not a well-known ssl port.

Reply:
`Requests` requires full address with schema specified, not atomic for each component. If we decouple it manually, we have to construct URL ourselves back, which I'm not sure about

Reply:
Couldn't we use [`urlunparse`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlunparse) to reconstruct the fully qualified service endpoint?

```python
urlunparse((self.scheme, self.hostname + ':' + str(self.port), self.path, None, None, None))
```

Reply:
Added some logic with urlparse/urlunparse, let me know what u think.